ui/src/app/machines/views/MachineDetails/MachineNetwork/AddInterface/AddInterface.tsx

```diff
@@ -0,0 +1,137 @@
+import { useCallback, useEffect } from "react";
+
+import { Spinner } from "@canonical/react-components";
+import { usePrevious } from "@canonical/react-components/dist/hooks";
+import { useDispatch, useSelector } from "react-redux";
+import * as Yup from "yup";
+
+import AddInterfaceFormFields from "./AddInterfaceFormFields";
+import type { AddInterfaceValues } from "./types";
+
+import FormCard from "app/base/components/FormCard";
+import FormCardButtons from "app/base/components/FormCardButtons";
+import FormikForm from "app/base/components/FormikForm";
+import { MAC_ADDRESS_REGEX } from "app/base/validation";
+import { actions as fabricActions } from "app/store/fabric";
+import fabricSelectors from "app/store/fabric/selectors";
+import { actions as machineActions } from "app/store/machine";
+import machineSelectors from "app/store/machine/selectors";
+import type { Machine, MachineDetails } from "app/store/machine/types";
+import {
+  NetworkInterfaceTypes,
+  NetworkLinkMode,
+} from "app/store/machine/types";
+import { getNextNicName } from "app/store/machine/utils";
+import type { RootState } from "app/store/root/types";
+import { actions as vlanActions } from "app/store/vlan";
+import vlanSelectors from "app/store/vlan/selectors";
+
+type Props = {
+  close: () => void;
+  systemId: MachineDetails["system_id"];
+};
+
+const InterfaceSchema = Yup.object().shape({
+  ip_address: Yup.string(),
+  mac_address: Yup.string()
+    .matches(MAC_ADDRESS_REGEX, "Invalid MAC address")
+    .required("MAC address is required"),
+  mode: Yup.mixed().oneOf(Object.values(NetworkLinkMode)),
+  name: Yup.string(),
+  fabric: Yup.number().required("Fabric is required"),
+  subnet: Yup.number(),
+  tags: Yup.array().of(Yup.string()),
+  vlan: Yup.number().required("VLAN is required"),
+});
+
+const AddInterface = ({ close, systemId }: Props): JSX.Element | null => {
+  const dispatch = useDispatch();
+  const machine = useSelector((state: RootState) =>
+    machineSelectors.getById(state, systemId)
+  );
+  const cleanup = useCallback(() => machineActions.cleanup(), []);
+  const errors = useSelector(machineSelectors.errors);
+  const fabrics = useSelector(fabricSelectors.all);
+  const fabricsLoaded = useSelector(fabricSelectors.loaded);
+  const creatingPhysical = useSelector((state: RootState) =>
+    machineSelectors.getStatusForMachine(state, systemId, "creatingPhysical")
+  );
+  const previousSaving = usePrevious(creatingPhysical);
+  const saved = !creatingPhysical && previousSaving && !errors;
```

Comment:
Could you use the `useMachineDetailsForm` hook here? It should handle errors, saving and saved state and close the form on saved.

Reply:
Yes! Thanks, I'll give it a go.